jekyll/_cci2/server/operator/manage-virtual-machines-with-vm-service.adoc

```diff
@@ -0,0 +1,128 @@
+= Manage virtual machines with VM Service
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+VM service controls how https://circleci.com/docs/2.0/configuration-reference/#machine[`machine`] executor and https://circleci.com/docs/2.0/building-docker-images[Remote Docker] jobs are run.
+
+This section describes the available configuration options for VM service. Refer to the default `values.yaml` file for details on how to pre-scale virtual machines.
+
+toc::[]
+
+CAUTION: We recommend that you leave these options at their defaults until you have successfully configured and verified the core and build services of your server installation. Steps to set up VM service are provided in the installation guide for https://circleci.com/docs/2.0/server/installation/phase-3-execution-environments/#aws[AWS] and https://circleci.com/docs/2.0/server/installation/phase-3-execution-environments/#gcp[GCP].
+
+WARNING: If https://circleci.com/docs/2.0/docker-layer-caching/[Docker Layer Caching (DLC)] is used, VM Service instances need to be spun up on demand. For this to happen, **either** ensure any preallocated instances are in use, **or** set both remote Docker and `machine` preallocated instance fields to `0`.
+
+CAUTION: When using preallocated instances be aware that a cron job is scheduled to cycle through these instances once per day to ensure they do not end up in an unworkable state.
+
+[#vm-provider]
+== VM provider
+The following configuration options are for the VM provider: either AWS or GCP.
+
+[#aws]
+=== AWS
+You will need to add a section to your `values.yaml` file to configure VM Service to work with AWS EC2. 
+
+[#aws-authentication]
+==== Authentication
+One of the following options is required:
+
+* Either, select "IAM Keys" and provide:
+** *Access Key ID* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[Access Key ID] for EC2 access.
+** *Secret Key* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[Secret Key] for EC2 access.
++
+NOTE: The Access Key and Secret Key used by VM Service differs from the policy used by object storage in the previous section.
+
+* Or, select "IAM role" and provide:
+** *Role ARN* (required): 
++
+https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html[Role ARN for Service Accounts] (Amazon Resource Name) for EC2 access.
+
+[source,yaml]
+----
+vm_service:
+  providers:
+    ec2:
+      enabled: true
+      region: <region>
+      # Subnets must be in the same availability zone
+      subnets:
+      - <subnet-id>
+      securityGroupId: <security-group-id>
+
+      # Authenticate with IAM access keys
+      accessKey: <access-key>
+      secretKey: <secret-key>
+      # or IRSA (IAM roles for service accounts)
+      irsaRole: <role-arn>
+----
+
+[#default-aws-ami-list]
+==== Default AWS AMI list
+
+The default AMIs for server v4.x are based on Ubuntu 20.04.
```

Comment:
@atulsingh0 should I update this list with: https://github.com/circleci/server/blob/2d49512c963a12c3fedf5a61b38e8d6352777bf1/templates/vm-service/configmap.yaml#L3 ? 
And should I add the lest below that for Arm?

Reply:
Or alternatively can server operators see that list in the codebase?